tests/kernel.rs

```diff
@@ -102,3 +120,473 @@ fn kernel() {
 
     assert!(res.data.is_none());
 }
+
+#[test]
+fn test_fixed_amount_splitter_local_with_message_and_funds() {
+    let InterchainTestEnv { juno, .. } = InterchainTestEnv::new();
+
+    let recipient = juno.chain.addr_make("recipient");
+
+    // Deploy on Osmosis
+    let splitter_juno = FixedAmountSplitterContract::new(juno.chain.clone());
+    splitter_juno.upload().unwrap();
+
+    splitter_juno
+        .instantiate(
+            &andromeda_finance::fixed_amount_splitter::InstantiateMsg {
+                recipients: vec![andromeda_finance::fixed_amount_splitter::AddressAmount {
+                    recipient: Recipient {
+                        address: AndrAddr::from_string(recipient.clone()),
+                        msg: None,
+                        ibc_recovery_address: None,
+                    },
+                    coins: vec![Coin {
+                        denom: juno.denom.clone(),
+                        amount: Uint128::new(100),
+                    }],
+                }],
+                default_recipient: None,
+                lock_time: None,
+                kernel_address: juno.aos.kernel.address().unwrap().into_string(),
+                owner: None,
+            },
+            None,
+            None,
+        )
+        .unwrap();
+
+    // Register contract
+    juno.aos
+        .adodb
+        .execute(
+            &os::adodb::ExecuteMsg::Publish {
+                code_id: splitter_juno.code_id().unwrap(),
+                ado_type: "fixed-amount-splitter".to_string(),
+                action_fees: None,
+                version: "1.0.0".to_string(),
+                publisher: None,
+            },
+            None,
+        )
+        .unwrap();
```

Comment:
We need a macro to do this ideally. It's a common step in testing so we should abstract it as much as possible.